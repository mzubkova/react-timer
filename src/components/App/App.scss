body {
  font-family: "Roboto", sans-serif;
}
.light {
  background-color: #dfe3eb;
  color: #525bc4;
}

.dark {
  background-color: #282c34;
  color: white;
}

.theme-toggle-btn {
  padding: 1%;
  border: 0;
  cursor: pointer;
  bottom: 5%;
  font-size: 0.5em;
  position: absolute;
}

.icon-light {
  color: #dfe3eb;
  background-color: #525bc4;
}

.icon-dark {
  background-color: #dc1568;
  color: white;
}
