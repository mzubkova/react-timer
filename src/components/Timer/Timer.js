import React, { useState, useContext } from "react";
import PropTypes from "prop-types";

import TimeDisplay from "./TimeDisplay";

import { ThemeContext } from "../Context/ThemeContext";

import "./Timer.scss";

const Timer = ({ initialState = false, initialTime = 0, step = 1000 }) => {
  const [time, setTime] = useState(initialTime);
  const [isActive, setIsActive] = useState(initialState);
  const [valueInput, setValueInput] = useState("");

  const { theme } = useContext(ThemeContext);

  const timerButtonClass = `timer-button timer-button--${theme}`;

  const handleValueInput = (e) => setValueInput(e.target.value);

  const timeToggle = () => setIsActive(!isActive);

  const resetTimer = () => setTime(0);

  const setTimer = () => {
    setTime(+valueInput);
    setValueInput("");
  };

  return (
    <div className={`timer ${theme}`}>
      <div className="timer-inner">
        <label>
          <input
            className={`timer-input timer-input--${theme}`}
            value={valueInput}
            onChange={handleValueInput}
            placeholder="write your value"
          />
        </label>
        <button
          className={timerButtonClass}
          onClick={setTimer}
          value={valueInput}
        >
          Set
        </button>
        <button className={timerButtonClass} onClick={timeToggle}>
          {isActive ? "Pause" : "Start"}
        </button>
        <button className={timerButtonClass} onClick={resetTimer}>
          {isActive} Reset
        </button>
        <TimeDisplay initialTime={time} step={step} isActive={isActive} />
      </div>
    </div>
  );
};

Timer.propTypes = {
  item: PropTypes.shape({
    initialState: PropTypes.bool,
    initialTime: PropTypes.number,
    step: PropTypes.number,
  }),
  handleValueInput: PropTypes.func,
  timeToggle: PropTypes.func,
  resetTimer: PropTypes.func,
  setTimer: PropTypes.func,
};

export default Timer;
