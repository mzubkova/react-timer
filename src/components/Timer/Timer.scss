.light {
  background-color: #dfe3eb;
  color: #525bc4;
}

.dark {
  background-color: #282c34;
  color: white;
}

.title {
  font-size: 1.5rem;
  font-weight: 500;
  color: #170621;
  text-align: center;
  margin: 30px 0 0;

  &--dark {
    color: #fff;
  }
}

.timer {
  align-content: center;
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  overflow: auto;
  position: fixed;
  width: 100%;
  &-inner {
    margin: 0 auto;
    text-align: left;
    width: 600px;
  }

  &-input {
    width: 100px;
    height: 30px;
    border: 2px solid #ff3cac;
    border-radius: 5px;
    background-color: #dfe3eb;

    &--dark {
      background-color: #282c34;
      color: #fff;
      &::placeholder {
        color: #fff;
      }
    }
  }
}

.timer-button {
  margin: auto 20px;
  padding: 10px;
  width: 80px;
  height: 80px;
  border-radius: 50%;
  outline: none;
  text-transform: uppercase;
  cursor: pointer;
  color: #fff;

  &:hover {
    background-color: #322a92;
  }

  &:active {
    background-color: #dd87e2;
  }

  &--light {
    border: 3px solid #4dbddc;
    background-color: #784ba0;
  }

  &--dark {
    background-color: #4dbddc;
    border: 3px solid #cf30c8;
    color: #000;
  }
}

.timer-progress {
  width: 100%;
  height: 50px;
  border: 1px solid #2bd2ff;
  border-radius: 10px;
  justify-self: flex-start;
  margin-top: 40px;
  background-color: #ff3cac;
  background-image: linear-gradient(
    225deg,
    #ff3cac 0%,
    #784ba0 50%,
    #2b86c5 100%
  );
}

.timer-progress > h3 {
  font-size: 1rem;
  font-weight: 400;
  text-align: left;
  margin: 15px;
}
