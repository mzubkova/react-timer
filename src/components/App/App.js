import React, { useState } from "react";

import Timer from "../Timer";
import Header from "../Header";

import { ThemeContext } from "../Context/ThemeContext";

const App = () => {
  const [theme, setTheme] = useState("light");
  const value = { theme, setTheme };
  return (
    <>
      <ThemeContext.Provider value={value}>
        <Header />
        <Timer />
      </ThemeContext.Provider>
    </>
  );
};

export default App;
