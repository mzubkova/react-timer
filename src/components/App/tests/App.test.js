import React from "react";
import { shallow, mount } from "enzyme";
import App from "../App";
import Timer from "../App";

describe("App", () => {
  it("Should match shapshot", () => {
    const component = shallow(<App />);
    expect(component.html()).toMatchSnapshot();
  });
  it("Should have Header", () => {
    const component = mount(<App />);
    expect(component.find("Header")).toHaveLength(1);
  });
  it("Should have Timer", () => {
    const component = mount(<App />);
    expect(component.find("Timer")).toHaveLength(1);
  });
});
