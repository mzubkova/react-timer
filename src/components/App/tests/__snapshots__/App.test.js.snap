// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match shapshot 1`] = `"<div class=\\"header header--light\\"><button class=\\"switcher-btn\\">Dark theme</button></div><div class=\\"timer light\\"><div class=\\"timer-inner\\"><label><input class=\\"timer-input timer-input--light\\" value=\\"\\" placeholder=\\"write your value\\"/></label><button class=\\"timer-button timer-button--light\\" value=\\"\\">Set</button><button class=\\"timer-button timer-button--light\\">Start</button><button class=\\"timer-button timer-button--light\\"> Reset</button><h1 class=\\"title title--light\\">Time: 00:00:00</h1><div class=\\"timer-progress\\" style=\\"width:NaN%\\"></div></div></div>"`;
